docs/source/format/CDeviceDataInterface.rst

```diff
@@ -650,6 +652,340 @@ The stream source is not assumed to be thread-safe. Consumers wanting to
 call ``get_next`` from several threads should ensure those calls are
 serialized.
 
+Async Device Stream Interface
+=============================
+
+The :ref:`C stream interface <_c-device-stream-interface>` provides a synchronous
+API centered around the consumer calling the callback functions to retrieve
```

Comment:
Are these really "call-back" functions? I know the spec and code calls them callbacks, but "producer functions" will be less confusing here.

```suggestion
API centered around the consumer calling the producer functions to retrieve
```